README.md

```diff
@@ -346,6 +346,28 @@ Contributions and suggestions for new features are welcome, as are bug reports!
 Please create a new [issue](https://github.com/ewels/rich-click/issues)
 or better still, dive right in with a pull-request.
 
+### Local setup
+
+1. Create a new venv with a python3.6+ interpreter using `python3 -m venv venv`
+2. Activate the venv with `source venv/bin/activate`
+3. Install our the package as an editable including all dev dependencies with `pip3 install -e ."[dev]"`
+4. Install pre-commit with `pre-commit install`
+
+#### Pre-commit
+
+Our pre-commit hooks contain the following hooks:
+
+- prettier: formats our yaml files nicely.
+- no relative imports: prevents you from using relative imports.
+- iSort: will automatically sort the imports alphabetically.
+- black: will automatically format your code to be according to standardized python format.
+- flake8: will do linting checks to make sure all your code is correctly styled and used.
+- mypy: static type checker which verifies you are not using objects incorrectly.
+
+As mentioned, some of these tools automatically fix your code while other only highlight potential issues.
+Therefore, sometimes it enough to try to commit a second time, and it will pass, while others will require
+manual changes to your code.
+
```

Comment:
```suggestion

In rare cases it may be difficult or undesirable to change to code to pass the linting rules.
If this happens, it's ok to add a flake8 `# noqa` or mypy `# type: ignore` comment to skip that line.
For details of how to do this, please see the [flake8 docs](https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors)
and [mypy docs](https://mypy.readthedocs.io/en/stable/common_issues.html#spurious-errors-and-locally-silencing-the-checker).
```